lib/public/components/Filters/common/filters/checkboxFilter.js

```diff
@@ -38,3 +38,24 @@ export const checkboxFilter = (name, values, isChecked, onChange, additionalProp
             for: `${name}Checkbox${value}`,
         }, value.toUpperCase()),
     ])));
+
+/**
+ * A general component for generating checkboxes.
+ * @param {SelectionModel} selectionModel fiter model
+ * @param {Object} [additionalProperties = {}] additional options that can be given to checkboxes.
+ * @param {string} [additionalProperties.id] input identifers prefix
+ * @return {Componenet} filter componenet
+ */
+export const modeledCheckboxFilter = (selectionModel, additionalProperties = {}) =>
+    h('.flex-row.flex-wrap', selectionModel.options.map((option) => h('.form-check.flex-grow', [
+        h('input.form-check-input', {
+            id: `${additionalProperties.id}-checkbox-${option.value}`,
```

Comment:
If it's optional you may need to check it. Anyway, we should avoid puting ids everywhere, and make more effort on tests to avoid using it...

Reply:
Still, you need to handle the undefined case

Reply:
handled, also id renamed to selector (as for tag details), because id might be misleading